llvm/lib/Demangle/MicrosoftDemangle.cpp

```diff
@@ -323,8 +335,30 @@ Demangler::demangleSpecialTableSymbolNode(std::string_view &MangledName,
   }
 
   std::tie(STSN->Quals, IsMember) = demangleQualifiers(MangledName);
-  if (!consumeFront(MangledName, '@'))
-    STSN->TargetName = demangleFullyQualifiedTypeName(MangledName);
+
+  NodeList *TargetCurrent = nullptr;
```

Comment:
Instead of building a linked list here, could we just put these into a SmallVector and create the NodeArray from that directly?